### PR TITLE
dev environment setup scripts update

### PR DESCRIPTION
**Pull request by dagar**
- Ubuntu install NuttX toolchain, gazebo, and switch to python3
 - OSX switch to python3

### CODE REVIEW

File: Tools/setup/ubuntu.sh [hunk 1]
```diff
@@ -50,33 +52,62 @@ sudo apt-get -yy --quiet --no-install-recommends install \
...
+	vim-common \
+	;
+
+NUTTX_GCC_VERSION=gcc-arm-none-eabi-7-2017-q4-major
+wget -O /tmp/${NUTTX_GCC_VERSION}-linux.tar.bz2 https://armkeil.blob.core.windows.net/developer/Files/downloads/gnu-rm/7-2017q4/${NUTTX_GCC_VERSION}-linux.tar.bz2 && \
+sudo tar -jxf /tmp/${NUTTX_GCC_VERSION}-linux.tar.bz2 -C /opt/
+
+exportline="export PATH=/opt/${NUTTX_GCC_VERSION}/bin:\$PATH"
+
+if grep -Fxq "$exportline" $HOME/.profile;
```
Comment (hamishwillee) [inline Tools/setup/ubuntu.sh]: Perhaps do this before installing the toolchain, and use it as a condition/test for installing the toolchain?

File: Tools/setup/ubuntu.sh [hunk 2]
```diff
@@ -50,33 +52,62 @@ sudo apt-get -yy --quiet --no-install-recommends install \
...
+then
+	echo "${NUTTX_GCC_VERSION} path already set.";
+else
+	echo $exportline >> $HOME/.profile;
+fi
+
+# remove user from dialout group
```
Comment (hamishwillee) [inline Tools/setup/ubuntu.sh]: This is adding  (not removing) a user to the dialout group, and requires a logout/login
We should prompt for that at end

**Comment by hamishwillee** [comment]
@dagar My interest in this is as a replacement for the ubuntu Devguide scripts. I have run the script in Ubuntu 16.04 (only) from the Firmware root using `source ./Tools/setup/ubuntu.sh`.
1. Installer went through to completion ([log gist here](https://gist.github.com/hamishwillee/b0c1ab8ad3e8482a59341b6a2da93a18)) with warning `There were unauthenticated packages and -y was used without --allow-unauthenticated`
1. Tried `make posix jmavsim` and gazebo. Fails with missing jinga2. Ubuntu suggests for debian systems try: `sudo apt-get install python-jinja2` and for all others try `sudo -H pip install jinja2`. Installed that using `sudo apt-get install python-jinja2
1. Then got error on missing em. But I can see empy in both pip list and pip3 list. Anyway, installed using `sudo apt-get install python-empy` only to get TOML error. Gave up. Thoughts?
1. Tried FMUv5 build - failed
```
ubuntu@ubuntu:~/github/Firmware$ make px4fmu-v5_default
Makefile:212:  px4fmu-v5_default has been deprecated and will be removed, please use px4_fmu-v5_default!
make  px4_fmu-v5_default
make[1]: Entering directory '/home/ubuntu/github/Firmware'
-- PX4 config file: /home/ubuntu/github/Firmware/boards/px4/fmu-v5/default.cmake
-- PX4 config: px4_fmu-v5_default
-- PX4 platform: nuttx
-- PX4 lockstep: disabled
-- PX4 version: v1.9.0-beta1-456-gadf9006
-- cmake build type: MinSizeRel
CMake Error at platforms/nuttx/cmake/Toolchain-arm-none-eabi.cmake:40 (message):
  could not find genromfs
Call Stack (most recent call first):
  /usr/share/cmake-3.5/Modules/CMakeDetermineSystem.cmake:98 (include)
  CMakeLists.txt:263 (project)
```

More generally
1. What about FastRTPS. Does this need to be installed?
2. We use current scripts as a "base" with further instructions for other platforms - e.g. https://dev.px4.io/en/setup/dev_env_linux_ubuntu.html#snapdragon-flight or Bebop or whatever. When working, should this still be a good base for those, or will there be conflicts?
3. Is it reasonable going forward to have a conditional flag for ROS setup, or is that a discussion for another day?

**Comment by dagar** [comment]
For Ubuntu 16.04 the problem (at least the first one) is the python dependencies. This is probably bad form, but I'm tempted to just install them all for both python2 and python3 until we can cleanly transition to python3 entirely.

> 1. What about FastRTPS. Does this need to be installed?

Nice to have, but not currently needed. Skipped for now because I wanted to look for clean packages.

> 2. We use current scripts as a "base" with further instructions for other platforms - e.g. https://dev.px4.io/en/setup/dev_env_linux_ubuntu.html#snapdragon-flight or Bebop or whatever. When working, should this still be a good base for those, or will there be conflicts?

Conflicts no, but it's currently more packages than we put into the base container (to keep the size down). Now that I think of it again I suppose we'll need to make the Nuttx and gazebo portions optionally so that the script can be used by PX4/containers directly.

> 3. Is it reasonable going forward to have a conditional flag for ROS setup, or is that a discussion for another day?

Avoiding the ROS version discussion for now, plus the actual ROS install is straightforward copy paste from their site.

**Comment by hamishwillee** [comment]
> For Ubuntu 16.04 the problem (at least the first one) is the python dependencies. This is probably bad form, but I'm tempted to just install them all for both python2 and python3 until we can cleanly transition to python3 entirely.

Works for me. Let me know when you want retest, as long as it is tomorrow :-)

All the extra bits should ideally be conditional from an end user point of view too - and in that case including FastRTPS also makes sense as an option. But it can be a "later" option.

>> Is it reasonable going forward to have a conditional flag for ROS setup, or is that a discussion for another day?
>> Avoiding the ROS version discussion for now, plus the actual ROS install is straightforward copy paste from their site.

ROS "vanilla" setup is indeed simple, but ROS and PX4 is a hell of a lot more difficult than that, in particular to get the ROS2 stuff working nicely. @TSC21 has some scripts we could leverage. Currently in discussion about what is really needed: https://github.com/PX4/Devguide/issues/764

**Comment by MaEtUgR** [comment]
Cool, that's the way to go. I can do some VM testing later on.

File: Tools/setup/ubuntu.sh [hunk 3]
```diff
@@ -50,33 +52,62 @@ sudo apt-get -yy --quiet --no-install-recommends install \
...
+	automake \
+	bison \
+	bzip2 \
+	flex \
+	gdb-multiarch \
+	gperf \
+	libncurses-dev \
+	libtool \
+	pkg-config \
+	vim-common \
```
[the remark this answers is not in the document]
Comment (dagar) [reply Tools/setup/ubuntu.sh]: Looks like xxd is a package in Ubuntu 18.04, but I don't recall seeing it in 16.04. We could expand the conditionals near the top for Ubuntu version differences.

**Pull request merged**